File: 백준/Silver/1181. 단어 정렬/단어 정렬.cc
#include<iostream>
#include<algorithm>
#include<vector>
using namespace std;
string arr[20005];
vector<string>v;

bool compare(const string& a, const string& b)
{
	if (a.length() != b.length()) {
		return a.length() < b.length();
	}
	return a < b;
}


int main()
{
	ios::sync_with_stdio(0); cin.tie(0);
	int n; cin >> n;

	for (int i = 0; i < n; i++) cin >> arr[i];

	sort(arr, arr + n, compare);
	v.push_back(arr[0]);

	for (int i = 1; i < n; i++) {
		if (v[v.size() - 1] == arr[i]) continue;
		else v.push_back(arr[i]);
	}
	for (auto x : v) cout << x << '\n';
	return 0;
}

File: 백준/Silver/1181. 단어 정렬/단어 정렬.java
import java.io.*;
import java.util.*;

class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		HashSet<String>set = new HashSet<>();
		
		while(n-- > 0){
			set.add(br.readLine());
		}
		
		int setSize = set.size();
		String[] arr = set.toArray(new String[setSize]);
		
		Arrays.sort(arr, new Comparator<String>(){
			@Override
			public int compare(String a, String b){
				if(a.length() == b.length()){
					return a.compareTo(b);
				}else{
					return a.length() - b.length();
				}
			}
		});
		
		StringBuilder sb = new StringBuilder();
		for(String s : arr) sb.append(s).append("\n");
		System.out.println(sb.toString());
	}
}